### ISSUE TITLE
Publish on Flathub please

### ISSUE THREAD

**Issue by alex285** [ubuntu/yaru#201]
..otherwise all Flatpak apps getting Adwaita theme!

**Comment by MrMarukesu** [comment]
> Flathub only hosts stable application releases, and not development snapshots. Manifests should therefore refer to tarballs or git tags and not the tip of a branch.
so, communitheme needs a stable relase first to be able do publish in flathub

**Comment by Feichtmeier** [comment]
@TingPing wow thank you! How does the updating works?

**Comment by TingPing** [comment]
It builds from the git repo so just bump the commit: https://github.com/flathub/org.gtk.Gtk3theme.Communitheme/blob/master/org.gtk.Gtk3theme.Communitheme.json#L60

**Comment by clobrano** [comment]
I guess flatpak version is a bit outdated :sweat_smile: 

![image](https://user-images.githubusercontent.com/2883614/41821941-88fe20f4-77e8-11e8-9c75-3a9fd9c84521.png)

is it possible to replace commit SHA with some alias like "HEAD"?

**Comment by TingPing** [comment]
> is it possible to replace commit SHA with some alias like "HEAD"?

No, all builds should be of a specific commit and updating is not automated.

> Otherwise someone (I can) should step up and updates the commit together with promotion of edge snap branch to stable

I have invited you to the repo with write access. Feel free to update the commit and let me know if you have any questions or somebody else wants access.

**Comment by clobrano** [comment]
Thanks @TingPing, could you give me some hints (never used flatpak before)? How's the workflow?
Once the .json is updated, an automatic build starts? Is it possilble to check the status?

**Comment by TingPing** [comment]
> Once the .json is updated, an automatic build starts? Is it possilble to check the status?

Yes every commit to master will trigger a build that can be viewed here: https://flathub.org/builds/#/

(github also shows an :heavy_check_mark: or :x: on the commit history that links to builds)

**Comment by TingPing** [comment]
If you want to debug any problems locally, which honestly shouldn't be too common for your usage, you can run it locally by installing `flatpak-builder` and running `flatpak-builder builddir org.gtk.Gtk3theme.Communitheme --force-clean`.

**Comment by clobrano** [comment]
And what about our shell? Do you think it is possible to have a flatpak for that?

**Comment by TingPing** [comment]
> And what about our shell? Do you think it is possible to have a flatpak for that?

So `flatpak` is more strict than `snap`, sandboxing isn't optional and it isn't intended to install files onto the host. You can do really naughty things and bypass the sandboxing but it is really not meant for that.

**Comment by galgalesh** [comment]
Just to give some context on how we use snap for the Gnome Shell theme.

Our theme is strictly sandboxed, a sandbox just doesn't really mean a lot for a theme since a theme doesn't execute anything, it's just a list case rules that the shell applies to itself.

The files of every snap are available on the filesystem in `/snap`, readable by all. So we tell the shell to look in that directory using the `XDG_DATA_DIRS` environment variable. This variable is set by the Communitheme session, which is shipped as a regular Ubuntu package and installed by default on any Ubuntu desktop install.

So we don't break out of the sandbox. The Communitheme snap doesn't even run anything, it's files are just used by the shell to theme itself.

Long story short; you could do the same with a flatpak, given that the files of an installed flatpak are available somewhere in the mounted filesystem. Is that the case?

A theme is basically a bunch of

**Comment by TingPing** [comment]
The files are generally readable in `~/.local/share/flatpak/app/org.example.Application/current/active/files/` or `/var/lib/flatpak/app/org.example.Application/current/active/files/`

**Comment by clobrano** [comment]
I know nothing about flatpak yet, sorry. I only thought that it would have been looked incomplete without the gnome shell part (not considering the icon part and the sound).

**Comment by TingPing** [comment]
The Flatpak packaged Gtk theme exists for Flatpak applications. The host is must get the theme via other means already.

**Comment by galgalesh** [comment]
> I will say that that is an unconventional use of Flatpak as just a data delivery method. I'm not sure it adds much value if there is a native package providing the session anyway. So I'm not sure if Flathub would want that hosted on it or not, it would be something for us to discuss at least if you really wanted to move forward with that.

It would give us the same benefits as the snap: easy delivery of updates controlled by us. However, since nobody asked for it, I won't pursue it further at this moment.

However, it's important that Flatpak apps are themed correctly with Communitheme so we should update the Flatpak theme together with the snap stable release.

I see that @clobrano did just that with the latest stable release, so that means this issue is fixed?

**Comment by clobrano** [comment]
> I see that @clobrano did just that with the latest stable release, so that means this issue is fixed?

Correct, I did it manually, but looking for a way to do it automatically (probably by some script with sed and git locally, still don't know)

**Comment by Feichtmeier** [comment]
Hi @TingPing 

After our name/repo change the flatpak theme is not updated anymore: https://github.com/ubuntu/yaru/issues/688

Could you look into that?

**Comment by TingPing** [comment]
Somebody made a new repo: https://github.com/flathub/org.gtk.Gtk3theme.Yaru

Can you clarify what your issue is exactly?

**Comment by Feichtmeier** [comment]
@TingPing oh that's funny. That somebody is @taciturasa who opened the other issue :)
So maybe, to deconstruct this, he opened the issue here, tried to fix it on that repo you posted, but forgot to report back what he did :man_shrugging: 
Then sorry for bothering you.

Yet, I think I found a typo here: https://github.com/flathub/org.gtk.Gtk3theme.Yaru/blob/master/org.gtk.Gtk3theme.Yaru.json#L50
@taciturasa  please enlighten us

**Comment by TingPing** [comment]
> Yet, I think I found a typo here

Fixed, though module names don't affect the output package.

**Comment by Feichtmeier** [comment]
Okay great - would be cool if someone could confirm if this is fixed for flatpaks or not (I don't have flatpak installed)

**Comment by taciturasa** [comment]
fast fingers and desperation to understand what i'm doing. sorry about that

**Comment by taciturasa** [comment]
Also: it looks like the flatpak does not, in fact, work. Seriously have no idea what I'm doing; as @TingPing told before, lots of the stuff themes do isn't documented, so I'm kinda lost.

**Comment by clobrano** [comment]
I am pretty sure at some point this was working, since I was using Builder to test Nautilus and the running application used Yaru. However not it is using Adwaita again

**Comment by clobrano** [comment]
Installed Gnome Calendar with flatpak. It does download Yaru (see logs below) automatically, however the app uses Adwaita

```
flatpak install flathub org.gnome.Calendar
Required runtime for org.gnome.Calendar/x86_64/stable (org.gnome.Platform/x86_64/3.28) is not installed, searching...
Found in remote flathub, do you want to install it? [y/n]: y
Installing: org.gnome.Platform/x86_64/3.28 from flathub
[####################] 10 delta parts, 80 loose fetched; 260652 KiB transferred in 62 seconds
Installing: org.freedesktop.Platform.VAAPI.Intel/x86_64/1.6 from flathub
[####################] 1 delta parts, 2 loose fetched; 2621 KiB transferred in 1 seconds
Installing: org.freedesktop.Platform.ffmpeg/x86_64/1.6 from flathub
[####################] 1 delta parts, 2 loose fetched; 2649 KiB transferred in 1 seconds
Installing: org.gtk.Gtk3theme.Yaru/x86_64/3.22 from flathub
[####################] 1 delta parts, 1 loose fetched; 1 KiB transferred in 0 seconds
Installing: org.gnome.Platform.Locale/x86_64/3.28 from flathub
[####################] 4 metadata, 1 content objects fetched; 14 KiB transferred in 0 seconds
Installing: org.gnome.Calendar/x86_64/stable from flathub
[####################] 1 delta parts, 6 loose fetched; 3822 KiB transferred in 2 seconds
Installing: org.gnome.Calendar.Locale/x86_64/stable from flathub
[####################] 3 metadata, 1 content objects fetched; 6 KiB transferred in 0 seconds
```

**Comment by taciturasa** [comment]
Only thing I can think of is that the repository combining messed with the build. @TingPing can you confirm that that could be?

**Comment by clobrano** [comment]
I made a diff with Pop-theme flatpak configuration (that works), and the first difference I saw is the runtime value. Yaru uses `org.freedesktop.Sdk`, while PopOS `org.freedesktop.Platform`

@taciturasa  Do you mind if I push this change directly on the repository?

**Comment by taciturasa** [comment]
Nope, no problem. Do whatever you need to get it running :)

**Comment by clobrano** [comment]
Ok done. I assume we need to wait for some CI system to rebuild the package

**Comment by TingPing** [comment]
What does `gsettings get org.gnome.desktop.interface gtk-theme` output?

**Comment by clobrano** [comment]
The output is "Yaru"

**Comment by TingPing** [comment]
Ok yea the built package is just empty, I'l look into it.

**Comment by TingPing** [comment]
This should fix it: https://github.com/flathub/org.gtk.Gtk3theme.Yaru/commit/b606be8aba3aa8e5be1f1cfcd7e8cff32d2eb63b

Its an ugly hack and you should add some build time options to disable the unused bits.

**Comment by TingPing** [comment]
> Awesome ! Is the type: "shell"; what is its the purpose?

It just runs arbitrary commands before building.

**Comment by clobrano** [comment]
It works now :+1: 

![image](https://user-images.githubusercontent.com/2883614/44615910-4b0aaa00-a845-11e8-8706-0e8cea20c1ad.png)

**Comment by TingPing** [comment]
Icon themes are always passed through into Flatpak and it doesn't relate to the Gtk theme.

**Comment by clobrano** [comment]
That's right, also PopOS does not set the icon theme correctly.

**Comment by aviwad** [comment]
Erm, this issue needs to be opened, because for some applications the theming is working (such as Lollypop) and for some, it isn't working (such as Epiphany Web Preview, and the default Python GTK template in Gnome Builder)

**Comment by 7im4pple** [comment]
Can anyone give me the command to install this..excuse my ignorance. And does it cover the Yaru dark theme also?

**Comment by TingPing** [comment]
@7im4pple It is automatically installed on any `flatpak install` or `flatpak update`. The dark variant is not packaged so that is probably your problem.
